pkg/resources/management.cattle.io/v3/setting/validator.go

```diff
@@ -69,71 +82,243 @@ func (v *Validator) Admitters() []admission.Admitter {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get Setting from request: %w", err)
 	}
+
 	switch request.Operation {
 	case admissionv1.Create:
-		if err := a.validateUserRetentionSettings(newSetting); err != nil {
-			return admission.ResponseBadRequest(err.Error()), nil
-		}
+		return a.admitCreate(newSetting)
 	case admissionv1.Update:
-		if err := a.validateUserRetentionSettings(newSetting); err != nil {
-			return admission.ResponseBadRequest(err.Error()), nil
-		}
-		if err := a.validateAgentTLSMode(*oldSetting, *newSetting); err != nil {
-			return admission.ResponseBadRequest(err.Error()), nil
-		}
+		return a.admitUpdate(oldSetting, newSetting)
+	default:
+		return admission.ResponseAllowed(), nil
 	}
-	return admission.ResponseAllowed(), nil
 }
 
-func (a *admitter) validateUserRetentionSettings(s *v3.Setting) error {
+func (a *admitter) admitCreate(newSetting *v3.Setting) (*admissionv1.AdmissionResponse, error) {
+	return a.admitCommonCreateUpdate(nil, newSetting)
+}
+
+func (a *admitter) admitUpdate(oldSetting, newSetting *v3.Setting) (*admissionv1.AdmissionResponse, error) {
 	var err error
 
-	switch s.Name {
-	case "disable-inactive-user-after":
-		if s.Value != "" {
-			_, err = validateDuration(s.Value)
+	switch newSetting.Name {
+	case AgentTLSMode:
+		err = a.validateAgentTLSMode(oldSetting, newSetting)
+	default:
+	}
+
+	if err != nil {
+		return admission.ResponseBadRequest(err.Error()), nil
+	}
+
+	return a.admitCommonCreateUpdate(oldSetting, newSetting)
+}
+
+func (a *admitter) admitCommonCreateUpdate(_, newSetting *v3.Setting) (*admissionv1.AdmissionResponse, error) {
+	var err error
+
+	switch newSetting.Name {
+	case DeleteInactiveUserAfter:
+		err = a.validateDeleteInactiveUserAfter(newSetting)
+	case DisableInactiveUserAfter:
+		err = a.validateDisableInactiveUserAfter(newSetting)
+	case UserLastLoginDefault:
+		err = a.validateUserLastLoginDefault(newSetting)
+	case UserRetentionCron:
+		err = a.validateUserRetentionCron(newSetting)
+	case AuthUserSessionTTLMinutes:
+		err = a.validateAuthUserSessionTTLMinutes(newSetting)
+	default:
+	}
+
+	if err != nil {
+		return admission.ResponseBadRequest(err.Error()), nil
+	}
+
+	return admission.ResponseAllowed(), nil
+}
+
+// validateAuthUserSessionTTLMinutes validates the auth-user-session-ttl-minutes setting
+// to make sure it's a positive integer and that duration is not greater than
+// {disable|delete}-inactive-user-after settings if they are set.
+// If it encounters an error fetching or parsing {disable|delete}-inactive-user-after settings
+// it logs but doesn't return the error to avoid rejecting the request.
+func (a *admitter) validateAuthUserSessionTTLMinutes(s *v3.Setting) error {
+	if s.Value == "" {
+		return nil
+	}
+
+	userSessionDuration, err := parseMinutes(s.Value)
+	if err != nil {
+		return field.TypeInvalid(valuePath, s.Value, err.Error())
+	}
+	if userSessionDuration < 1 {
+		return field.TypeInvalid(valuePath, s.Value, "negative value")
```

Comment:
nit: might confuse people that reads `negative` as meaning strictly less than zero. Also I think it's more common to see `<= 0` instead of `< 1`.

```suggestion
		return field.TypeInvalid(valuePath, s.Value, "non-positive value")
```

Reply:
Resolving as we're allowing 0 now.

Reply:
Agreed. But it's back to being strictly checking for a negative.